artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/ConcurrentLongHashMap.java

```diff
@@ -457,10 +457,10 @@ private void rehash() {
             }
          }
 
-         capacity = newCapacity;
          keys = newKeys;
          values = newValues;
          usedBuckets = size;
+         capacity = newCapacity;
```

Comment:
Please check if is possible to use a lazySet too (maybe with an `AtomicFieldUpdater` or similar) to write release the keys/values in order to avoid a full barrier